### DISCUSSION TITLE
API help (v: 4.2.1), Inventory save problem

### DISCUSSION THREAD

**Discussion by GregorSomething** [Jikoo/OpenInv#98] [Q&A] [answered]
I have openinvapi version 4.2.1 as a dependnecy.
```xml
        <!-- OpenInv -->
        <dependency>
            <groupId>com.github.jikoo.OpenInv</groupId>
            <artifactId>openinvapi</artifactId>
            <version>4.2.1</version>
            <scope>provided</scope>
        </dependency>
```

I want to move items from one player to other, but i can not save the modification that i have made.
I will send my code here.
```java
    private IOpenInv api;
    public void transferItems(UUID from, UUID to) { // TODO Test
        Player fromPlayer = this.api.loadPlayer(Bukkit.getOfflinePlayer(from));
        Player toPlayer = this.api.loadPlayer(Bukkit.getOfflinePlayer(to));
        try {
            ISpecialPlayerInventory invFrom = this.api.getSpecialInventory(fromPlayer, false);
            ISpecialPlayerInventory invTo = this.api.getSpecialInventory(toPlayer, false);
            for (ItemStack item : invFrom.getBukkitInventory().getStorageContents()) {
                if (item != null) {
                    invTo.getBukkitInventory().addItem(item);
                }
            }
            invFrom.getBukkitInventory().clear();
        } catch (InstantiationException e) {
            throw new RuntimeException(e);
        }
```
I have tested it multiple times, but can not find save method.

**Comment by Jikoo** [comment, marked as answer]
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/Player.html#saveData()

**Reply by GregorSomething** [reply]
Am i doing something else wrong then, i can not seem to clear first ones inventory and save the content to others inventory?

**Reply by Jikoo** [reply]
Would recommend you just access the inventory directly via Player#getInventory - the SpecialPlayerInventory is really there for the purpose of remapping slots to display prettily - but other than that I believe what you're doing should work. Not exactly in a position to test for a while though, unfortunately.

**Reply by GregorSomething** [reply]
Ty, using Player#getInventory worked.
